src/Filecoin/Address.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright © 2017-2020 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Base32.h"
+#include "../Data.h"
+
+using namespace TW;
+using namespace TW::Filecoin;
+
+static const char BASE32_ALPHABET_FILECOIN[] = "abcdefghijklmnopqrstuvwxyz234567";
+static constexpr size_t checksumSize = 4;
+
+bool Address::isValid(const Data& data) {
+    if (data.size() < 2) {
+        return false;
+    }
+    if (!getType(data[0])) {
+        return false;
+    }
+    Type type = *getType(data[0]);
+    if (type == Type::ID) {
+        // Verify varuint encoding
+        if (data.size() > 11)
+            return false;
+        if (data.size() == 11 && data[10] > 0x01)
+            return false;
+        int i;
+        for (i = 1; i < data.size(); i++)
+            if ((data[i] & 0x80) == 0)
+                break;
+        return i == data.size() - 1;
+    } else {
+        return data.size() == 1 + Address::payloadSize(type);
+    }
+}
+
+static bool isValidID(const std::string& string) {
+    if (string.length() > 22)
+        return false;
+    for (int i = 2; i < string.length(); i++) {
+        if (string[i] < '0' || string[i] > '9') {
+            return false;
+        }
+    }
+    try {
+        size_t chars;
+        std::stoull(string.substr(2), &chars);
```

Comment:
Codacy complains about the unused return value, while the logic makes perfect sense.  I can see no trivial workaround, if you can't find some tricky one, we may need to disable this codacy rule for this source file.

Reply:
I'll try `[[maybe_unused]]` from C++17, I can't think of any alternative.